File: n3/3.php
<?php
echo "Введите количество чисел и желаемые числа: ";
$input = explode(" ", trim(fgets(STDIN)));
$amount = $input[0]; // количество чисел
$numbers = array_slice($input, 1); // список чисел

$odd = []; // список нечётных цифр
$count = 0; // счётчик нечётных цифр

// перебор чисел
foreach ($numbers as $num) {
    foreach (str_split($num) as $digit) {
        if ((int)$digit % 2 === 1) { // проверка на нечётность
            $count++;
        }
    }
    $odd[] = $count; // добавление в список
    $count = 0; // сброс счётчика
}

echo "Количество нечётных цифр в числах: " . implode(" ", $odd) . "\n";
?>

File: n3/3.rs
use std::io;
fn main() {
    println!("Введите количество чисел и желаемые числа: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();

    // Преобразуем ввод в вектор
    let input: Vec<&str> = input.trim().split_whitespace().collect();
    let amount: usize = input[0].parse().unwrap();
    let numbers = &input[1..];

    let mut odd = Vec::new(); // список нечётных цифр
    let mut count = 0; // счётчик нечётных цифр

    // перебор чисел
    for num in numbers {
        for digit in num.chars() {
            if digit.to_digit(10).unwrap() % 2 == 1 { // проверка на нечётность
                count += 1;
            }
        }
        odd.push(count); // добавляем количество нечётных цифр
        count = 0; // сброс счётчика
    }
    println!("Количество нечётных цифр в числах: {:?}", odd);
}
